Create account webview (#10)
Infomaniak/android-login: README.md

## Install

In the `build.gradle` of your lib/app (not root level), add this instruction :

```groovy
implementation 'com.github.Infomaniak:android-login:2.6.1'
```

## Use Login

In the `onCreate` method of your Activity/Fragment, instantiate the library :

```kotlin
val loginInstance = InfomaniakLogin(
    context = this,
    clientId = CLIENT_ID,
    redirectUri = "$APP_UID:/oauth2redirect"
)
```

with these arguments :

- `context` : The current activity/fragment context (will be used to show a Chrometab)
- `clientId` : The client ID of the app
- `redirectUri` : The redirection URL after a successful login (in order to handle the codes)

**Once instantiated, you may use this everywhere in your activity/fragment, for example in a `onClick` method :**

```kotlin
buttonConnect.setOnClickListener {
    loginInstance.start()
}
```

> It'll create a custom chrome tab or launch the browser.

Reviewer: ```suggestion
It will create a custom chrome tab or launch the browser.
```

If needed (to see if an error occurred), the `start()` method returns a boolean indicating the success (or not) of the operation.

**Or you can use the `loginInstance` to create a login from a webview.**

```kotlin
buttonConnect.setOnClickListener {
    infomaniakLogin.startWebViewLogin(/*resultLauncher: ActivityResultLauncher<Intent>*/)
}
```

> It'll create an activity that contains a webview. The `startWebViewLogin` takes as parameter a resultLauncher to handle the

Reviewer: ```suggestion
It will create an activity that contains a WebView. The `startWebViewLogin` takes as parameter a resultLauncher to handle the
```

activity results.

```kotlin
private val webViewLoginResultLauncher = registerForActivityResult(StartActivityForResult()) { result ->
    with(result) {
        if (resultCode == RESULT_OK) {
            val authCode = data?.getStringExtra(InfomaniakLogin.CODE_TAG)
            val translatedError = data?.getStringExtra(InfomaniakLogin.ERROR_TRANSLATED_TAG)
            when {
                translatedError?.isNotBlank() == true -> showError(translatedError)
                authCode?.isNotBlank() == true -> authenticateUser(authCode)
                else -> showError(getString(RCore.string.anErrorHasOccurred))
            }
        }
    }
}
```

> This allows you to retrieve the answers from the webview, either the `code` for `auth` in case of success, or in case of error, we

Reviewer: ```suggestion
This allows you to retrieve the answers from the WebView, either the `code` for `auth` in case of success, or in case of error, we
```

retrieve the `error code` and the `translated error message`.

> ```xml

Reviewer: ```suggestion
Add this to the manifest, to authorize the WebView :

```xml
```

<activity android:name="com.infomaniak.lib.login.WebViewLoginActivity" />
```

> Add this to the manifest, to allow the webview
> 
> ## Use Create account

Reviewer: ```suggestion
## Use Create account
```

> **From your `loginInstance` you need to call the methode `startCreateAccountWebView`**

Reviewer: ```suggestion
**From your `loginInstance`, you need to call the method `startCreateAccountWebView`**
```

    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and

>     limitations under the License.

Reviewer: Are you sure about this change ?

I'd prefer to revert it, please.
